registry/config.go

```diff
@@ -289,6 +289,10 @@ func (config *serviceConfig) isSecureIndex(indexName string) bool {
 // where the `host` part can be either a domain name or an IP address. If it is a domain name, then it will be
 // resolved to IP addresses for matching. If resolution fails, false is returned.
 func isCIDRMatch(cidrs []*registry.NetIPNet, URLHost string) bool {
+	if len(cidrs) == 0 {
+		return false
+	}
```

Comment:
Sadly this currently doesn't help, as we're unconditionally initialising `RepoInfo` with localhost CIDRs; while that's relevant for _some_ uses of `RepoInfo`, it's not for others, and I'll look for follow-ups to split those use-cases.